### PR TITLE
t/168: Provided a static `Rect#getDomRangeRects` method for external usage

### LINKED ISSUE #168: Provide a static Rect#getDomRangeRects method for external usage
ATM there's no way to obtain all DOM Range rects e.g. when the range spans multiple lines of text using the `Rect` class. While it's possible to iterate over the list returned by `domRange.getClientRects()` and create a new `Rect` instance from each rect, such approach does not cover very specific cases which are **already handled** by the `Rect` helper class e.g. when there's no rect (Safari) etc..

So to use the smart logic of the `Rect` class and avoid code duplication, a static method is required. The very first use case is in https://github.com/ckeditor/ckeditor5-ui/issues/263, where depending on the direction of the seleciton, a proper DOM Range rect must be returned (either the first or the last one). Without the smart `Rect` logic, `ContextualToolbar` fails to show when the selection is collapsed in Safari and in other edge cases.

### PR DESCRIPTION
**Pull request by oleq**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Other: Provided a static `Rect#getDomRangeRects` method for external usage. Closes #168.

### CODE REVIEW

**Comment by Reinmar** [comment]
I don't understand these changes :D Waiting for more related PRs to see the reason for it.

File: src/dom/rect.js [hunk 1]
```diff
@@ -55,25 +55,7 @@ export default class Rect {
 		if ( isElement( source ) ) {
 			copyRectProperties( this, source.getBoundingClientRect() );
 		} else if ( isRange( source ) ) {
-			// Use getClientRects() when the range is collapsed.
-			// https://github.com/ckeditor/ckeditor5-utils/issues/153
-			if ( source.collapsed ) {
```
Comment (Reinmar) [inline src/dom/rect.js]: Where did that go?

Comment (oleq) [inline src/dom/rect.js]: It has been unified in `getDomRangeRects`. There's no need to differentiate it anymore.

**Comment by Reinmar** [comment]
> I don't understand these changes :D Waiting for more related PRs to see the reason for it.

@oleq explained this to me that there are some features which want to arbitrarily choose rects from the native range and based on that create `Rect` instances. But if they take one native rect and pass it to `Rect()` then that special logic of `Rect()` constructor which checks whether the range is collapsed etc wasn't executed.

**Pull request merged**